### PR TITLE
Make react-native-material-ui compatible with react-native-web

### PR DESCRIPTION
**Pull request by dalcib**
Demo app in [https://github.com/dalcib/react-native-material-ui-web-app](https://github.com/dalcib/react-native-material-ui-web-app)

### CODE REVIEW

File: src/styles/getPlatformElevation.js [hunk 1]
```diff
@@ -5,7 +5,40 @@ import { black } from './colors';
...
+    '0px 9px 11px -5px rgba(0, 0, 0, 0.2), 0px 18px 28px 2px rgba(0, 0, 0, 0.14), 0px 7px 34px 6px rgba(0, 0, 0, 0.12);',
+    '0px 9px 12px -6px rgba(0, 0, 0, 0.2), 0px 19px 29px 2px rgba(0, 0, 0, 0.14), 0px 7px 36px 6px rgba(0, 0, 0, 0.12);',
+    '0px 10px 13px -6px rgba(0, 0, 0, 0.2), 0px 20px 31px 3px rgba(0, 0, 0, 0.14), 0px 8px 38px 7px rgba(0, 0, 0, 0.12);',
+    '0px 10px 13px -6px rgba(0, 0, 0, 0.2), 0px 21px 33px 3px rgba(0, 0, 0, 0.14), 0px 8px 40px 7px rgba(0, 0, 0, 0.12);',
+    '0px 10px 14px -6px rgba(0, 0, 0, 0.2), 0px 22px 35px 3px rgba(0, 0, 0, 0.14), 0px 8px 42px 7px rgba(0, 0, 0, 0.12);',
+    '0px 11px 14px -7px rgba(0, 0, 0, 0.2), 0px 23px 36px 3px rgba(0, 0, 0, 0.14), 0px 9px 44px 8px rgba(0, 0, 0, 0.12);',
+    '0px 11px 15px -7px rgba(0, 0, 0, 0.2), 0px 24px 38px 3px rgba(0, 0, 0, 0.14), 0px 9px 46px 8px rgba(0, 0, 0, 0.12);'
+];
+
 const getPlatformElevation = (elevation) => {
```
Comment (chirag04) [inline src/styles/getPlatformElevation.js]: if we are doing i would say maybe use platform extensions?

`getPlatformElevation.web.js` `getPlatformElevation.ios.js` etc

**Comment by chirag04** [comment]
lgtm

**Comment by jaulz** [comment]
@dalcib does the ripple effect work for you as well? Currently, all components that make use of RippleEffect render a TouchableOpacity with activeOpacity 0.2 and focusedOpacity 0.7 (default props of react-native-web's TouchableOpacity @ https://github.com/necolas/react-native-web/blob/master/src/components/Touchable/TouchableOpacity.js). This looks a bit strange (element becomes transparent when focusing) so I guess we need to change the activeOpacity and focusedOpacity to 1 to prevent this.

**Comment by piranna** [comment]
I'm getting the next errors related to the split of the `getPlatformElevation` file:

```
Error in ./~/react-native-material-ui/src/styles/getTheme.js
Module not found: ./getPlatformElevation in /home/piranna/Trabajo/Quantum/flowtag_react/node_modules/react-native-material-ui/src/styles

 @ ./~/react-native-material-ui/src/styles/getTheme.js 1:677-710

Error in ./~/react-native-material-ui/src/Button/Button.react.js
Module not found: ../styles/getPlatformElevation in /home/piranna/Trabajo/Quantum/flowtag_react/node_modules/react-native-material-ui/src/Button

 @ ./~/react-native-material-ui/src/Button/Button.react.js 1:1356-1397

Error in ./~/react-native-material-ui/src/ActionButton/ActionButton.react.js
Module not found: ../styles/getPlatformElevation in /home/piranna/Trabajo/Quantum/flowtag_react/node_modules/react-native-material-ui/src/ActionButton

 @ ./~/react-native-material-ui/src/ActionButton/ActionButton.react.js 1:1462-1503
```

How can I be able to fix them?

**Comment by piranna** [comment]
By copying `getPlatformElevation.web.js` to `getPlatformElevation.js` it works on the web platform (`react-native-web`), probably it would makes sense to add a `getPlatformElevation.js` file that detects the platform and require one or another as a fallback, similar to what's done with other multi-platform libraries, what do you think?

I'll try to do a pull-request when I have some time for it.

**Comment by dalcib** [comment]
@jaulz, I'm not able to test this for now because I am traveling, but I will see since I can.

@piranna, if you aren't using `create-react-app` then you need to use the code above in your `webpack.config.js`

```
...
  resolve: {
    // `web` extension prefixes have been added for better support
    // for React Native Web.
    extensions: ['.web.js', '.js', '.json', '.web.jsx', '.jsx'],
...
```

**Comment by piranna** [comment]
> @piranna, if you aren't using create-react-app then you need to use the code above in your webpack.config.js
> 
> ...
>   resolve: {
>     // `web` extension prefixes have been added for better support
>     // for React Native Web.
>     extensions: ['.web.js', '.js', '.json', '.web.jsx', '.jsx'],
> ...

Ok, that worked, thanks :-) I'm somewhat new to the react environment at all :-)

**Pull request merged**